### PR TITLE
Write NFRs and combine with code standards

### LINKED ISSUE #3301: Non-functional requirements need documenting
**Is your feature request related to a problem? Please describe.**
As a contributor the requirements for merge might be unclear

**Describe the solution you'd like**
Probably as part of our contributing.md we must define clear non-functional requirements

**Describe alternatives you've considered**
Could put it in code standards

**Testing notes**
No testing needed

**Additional context**
Suggested content...

Before Merge
Unit Testing:
- Sufficient unit testis coverage for any new feature, route or page type
- A11y tests/swarms (cannot go live on www.bbc.com) without this (this would be ideal but not required at this stage)
Documentation:
- Any doc updates/ new documentation for a given page type/feature
E2E Testing:
- E2E service config added for a new service/updated for an existing service
- E2E should be updated and running against localhost and disabled for test and live
Promoting to test (www.test.bbc.com):
- All of the above
- Any new data endpoints need to be available on test (speak to core pod about enabling these)
- Raise an issue in the is-infra repo requesting the STM change for going to test
- Ensure any new page type is being tested by lighthouse
Once on test:
- E2E config must be updated immediately to ensure the e2e’s are running against the test environment

Promoting to Stage:
- All of the above
- Raise an issue in the is-infra repo requesting the STM change for going to stage

Promoting to Live:
- All of the above
- No outstanding a11y issues/swarms/tests (these will block going live)
- Raise an issue in the is-infra repo requesting the STM change for going to live

### PR DESCRIPTION
**Pull request by jamesbhobbs**
Resolves #3301

**Overall change:** Adds NFRs and revises existing code standards doc for understandability

---

- [x I have assigned myself to this PR and the corresponding issues
- [ ] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] This PR requires manual testing

### CODE REVIEW

File: docs/Code-Standards.md [hunk 1]
```diff
@@ -8,61 +8,78 @@ Please familiarise yourself with our:
...
+- If you have paired on a PR you should co-author commits.
 - If latest has a broken build, stop what you are doing and fix it.
-- Code reviews take precedence over implementation
-- Accessibility swarms and UX reviews are required for all front-end components
-
-#### Guideline Exceptions
-
-The coding standards are only guidelines and exceptions can be considered depending on the case.
+- Code reviews take precedence over implementation.
+- Accessibility swarms and UX reviews are required for all front-end components - although they are optional for components with alpha version numbers.
```
Comment (dr3) [inline docs/Code-Standards.md:40]: Maybe something should be said about using alpha components in simorgh

File: docs/Code-Standards.md [hunk 2]
```diff
@@ -8,61 +8,78 @@ Please familiarise yourself with our:
...
-- Code duplications warnings from Code Climate should be considered on a case-by-case basis (Readability > DRY).
+### Before Merge
 
-- If you have paired on a PR, this can include the person paired with but the reviewer must be a code maintainer
-
-- Alpha components do not require an immediate accessibility swarm or UX review.
+#### Unit Testing:
+- Unit testing is a mixture of Jest snapshot testing and enzyme assertion testing.
+- We push automated testing as far down the testing pyramid as possible. This means that if something can be tested via unit tests, to ensure we have a test coverage of > 80% on this repo.
+- For strictly **presentational** components Snapshots tests must be present and assertion testing is desirable.
```
Comment (dr3) [inline docs/Code-Standards.md]: >strictly **presentational** components
Should be in psammead surely?

File: docs/Code-Standards.md [hunk 3]
```diff
@@ -8,61 +8,78 @@ Please familiarise yourself with our:
...
+- For **containers** or otherwise logic handling code snapshot testing is not desirable and assertion testing is mandatory.
+- A11y tests/swarms (this would be ideal but not required at this stage)
+#### Documentation:
+- Any documentation updates and new documentation for a given page type or feature should be added in the PR that makes the changes.
+#### E2E Testing:
```
Comment (dr3) [inline docs/Code-Standards.md:55]: Worth mentioning that e2e's are mandatory for all simorgh routes, and then something about what is smoke tested

File: docs/Code-Standards.md [hunk 4]
```diff
@@ -8,61 +8,78 @@ Please familiarise yourself with our:
...
+- E2E service config added for a new service/updated for an existing service
+- E2E should be updated and running against localhost and disabled for test and live
+
+### Only applicable to BBC staff (inline references and links might not be accessible to non-BBC staff)
```
Comment (sareh) [inline docs/Code-Standards.md]:
```suggestion
### Only applicable to BBC staff
```

File: docs/Code-Standards.md [hunk 5]
```diff
@@ -8,61 +8,78 @@ Please familiarise yourself with our:
...
+
+#### Promoting to test (www.test.bbc.com):
+- Manually check all requirements for merge have been met.
+- Ensure any new page type is being tested by a11y and lighthouse.
+- Any new data endpoints need to be available on test (speak to core pod about enabling these).
+- Raise an issue in the [simorgh-infrastructure](https://github.com/bbc/simorgh-infrastructure/) repo requesting the STM change for going to test
```
Comment (sareh) [inline docs/Code-Standards.md]:
```suggestion
- Raise an issue to request the traffic manager (STM) change for going to test in the [simorgh-infrastructure repo](https://github.com/bbc/simorgh-infrastructure/). Note, this is a private repo only accessible to BBC staff. 
```

File: docs/Code-Standards.md (hunk 1 above)
Comment (jamesbhobbs) [reply docs/Code-Standards.md:40]: I think not using them in simorgh should be pretty obvious, but I have another issue for documenting how we use them anyhow: https://github.com/bbc/psammead/issues/1838

File: docs/Code-Standards.md (hunk 2 above)
Comment (jamesbhobbs) [reply docs/Code-Standards.md]: this is intended to be one doc for both repos

Comment (jamesbhobbs) [reply docs/Code-Standards.md]: Will make that clearer in https://github.com/bbc/psammead/issues/1838

File: docs/Code-Standards.md [hunk 6]
```diff
@@ -8,61 +8,78 @@ Please familiarise yourself with our:
...
 - AMP validation for pages
 - BBC a11y validation for pages
 - Lighthouse validation
 - Code Climate Quality tests
 
-#### Manual Processes
+## Manual Processes
 
+- Open your PRs as early as possible, as draft PRs.
+- Add your issues and PRs to the project "Simorgh".
```
Comment (amywalkerdev) [inline docs/Code-Standards.md:35]: Worth mentioning that they should add the relevant label for whichever stream of work it is for?

Comment (jamesbhobbs) [reply docs/Code-Standards.md:35]: Yes please write that as a suggestion @amywalker1987

File: docs/Code-Standards.md (hunk 3 above)
Comment (jamesbhobbs) [reply docs/Code-Standards.md:55]: @dr3 done

File: docs/Code-Standards.md (hunk 6 above)
Comment (amywalkerdev) [inline docs/Code-Standards.md:35]:
```suggestion
- Add your issues and PRs to the project "Simorgh".
- Add labels to your issues and PRs to identify which pod/stream of work they lie with. e.g "simorgh-core-stream"
```

**Review by amywalkerdev — CHANGES_REQUESTED** [review]

File: docs/Code-Standards.md (hunk 6 above)
Comment (jamesbhobbs) [reply docs/Code-Standards.md:35]: Decided to rewrite this a little

Comment (jamesbhobbs) [reply docs/Code-Standards.md:35]: Done

**Review by jroebu14 — APPROVED** [review]
Looks good

**Review by amywalkerdev — APPROVED** [review]
Looks good 👍

**Review by sareh — APPROVED** [review]
👍 Thanks for the update.

**Pull request merged**